src/humancellatlas/data/metadata/helpers/staging_area.py

```diff
@@ -217,37 +213,18 @@ def _entity_ids_by_type(self,
         return entity_ids
 
 
+JSON_FILE = TypeVar('JSON_FILE', bound=JsonFile)
+
+
 @attr.s(frozen=True, kw_only=True, auto_attribs=True)
 class GitHubStagingAreaFactory:
-    repo: Repository
```

Comment:
It is difficult to understand what each of these mean. It is not clear what the relationship between `location` and `path` is, since both are path-like. It is not clear what `url` is. Document each attribute with Sphinx syntax (`#:`) and consider better names. Use consistent names for variables and attributes. For example, below you say `url=clone_url` and that is a smell.